src/GraphQL/Execution/ExecutionStrategy.cs

```diff
@@ -51,7 +50,7 @@ await ExecuteNodeTreeAsync(context, rootNode)
         /// <summary>
         /// Builds the root execution node.
         /// </summary>
-        protected static RootExecutionNode BuildExecutionRootNode(ExecutionContext context, IObjectGraphType rootType)
+        protected RootExecutionNode BuildExecutionRootNode(ExecutionContext context, IObjectGraphType rootType)
```

Comment:
Virtual?

Reply:
Maybe. Note that other Build method is `static`. I still can't decide whether to make both static or both instance and (virtual).

Reply:
I think we should do private or protected virtual. If we do protected, then it breaks compatibility to make it virtual. In general I think many of these methods should be protected virtual instead of static, for the same reason. 

Also might consider moving the default ShouldIncludeNode implementation into ExecutionStrategy.ShouldIncludeNode

As much as I like interfaces, the execution strategy could be something where the interface really only helps for mocking and testing but in real life is not beneficial, assuming we use all virtual methods so it can be completely customized.

Reply:
(I am not suggesting we remove the IExecutionStrategy interface)

Reply:
> Also might consider moving the default ShouldIncludeNode implementation into ExecutionStrategy.ShouldIncludeNode

Already done.

Reply:
Done